jgrapht-guava/src/main/java/org/jgrapht/graph/guava/MutableValueGraphAdapter.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * (C) Copyright 2018-2018, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph.guava;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.function.ToDoubleFunction;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphType;
+import org.jgrapht.util.TypeUtil;
+
+import com.google.common.graph.EndpointPair;
+import com.google.common.graph.Graphs;
+import com.google.common.graph.MutableValueGraph;
+import com.google.common.graph.ValueGraph;
+import com.google.common.graph.ValueGraphBuilder;
+
+/**
+ * A graph adapter class using Guava's {@link MutableValueGraph}.
+ *
+ * <p>
+ * The class uses a converter from Guava's values to JGraphT's double weights. This is an one-way
+ * conversion meaning that calling {@link #setEdgeWeight(EndpointPair, double)} will throw an
+ * unsupported operation exception. Adjusting the weights can be done directly on the underlying
+ * Guava {@link ValueGraph} and propagated upstream using the provided value converter.
+ * 
+ * @author Dimitrios Michail
+ *
+ * @param <V> the graph vertex type
+ * @param <W> the value type
+ */
+public class MutableValueGraphAdapter<V, W>
+    extends BaseValueGraphAdapter<V, W, MutableValueGraph<V, W>>
+    implements Graph<V, EndpointPair<V>>, Cloneable, Serializable
+{
+    private static final long serialVersionUID = -5095044027783397573L;
+
+    protected W defaultValue;
+
+    /**
+     * Create a new adapter.
+     * 
+     * @param valueGraph the value graph
+     * @param defaultValue a default value to be used when creating new edges
+     * @param valueConverter a function that converts a value to a double
+     */
+    public MutableValueGraphAdapter(
+        MutableValueGraph<V, W> valueGraph, W defaultValue, ToDoubleFunction<W> valueConverter)
+    {
+        super(valueGraph, valueConverter);
+        this.defaultValue = Objects.requireNonNull(defaultValue);
```

Comment:
That was also my initial thought, until I realized that there is no `addEdge` method!

Reply:
@jsichi My understanding is that `ValueGraph.edgeValue(u, v)` returns a non-empty Optional if there is an edge between u and v, otherwise `Optional.empty()`. (That is, if there is no edge between u and v, it doesn't throw an exception; it just returns `Optional.empty()` instead.)

The reason I think this is because `MutableValueGraph` has only one method that adds edges: `putEdgeValue`, and AFAICT one has to provide an edge object to that method, in turn meaning that every edge in a `ValueGraph` has an associated edge object.

I could be wrong about this, though. :)